pkg/common/vclib/storagepod.go

```diff
@@ -202,3 +284,46 @@ func (sp *StoragePod) ListFirstClassDisks(ctx context.Context) ([]*FirstClassDis
 
 	return objs, nil
 }
+
+// GetFirstClassDiskByName gets a specific first class disks (FCD) on this datastore backed by this StoragePod
+func (sp *StoragePod) GetFirstClassDiskByName(ctx context.Context, diskName string) (*FirstClassDisk, error) {
+	err := sp.PopulateChildDatastores(ctx, false)
+	if err != nil {
+		glog.Errorf("PopulateChildDatastores failed. Err: %v", err)
+		return nil, err
+	}
+
+	m := vslm.NewObjectManager(sp.Datacenter.Client())
+
+	for _, child := range sp.Datastores {
+		oids, err := m.List(ctx, child)
+		if err != nil {
+			glog.Errorf("Failed to list disks. Err: %v", err)
+			return nil, err
+		}
+
+		var ids []string
```

Comment:
Could drop this var and loop, just use `oids` as-is in the main loop.

Reply:
See previous comment